src/kv.c

```diff
@@ -1178,16 +1181,31 @@ kvStore_WatchAll(kvWatcher **new_watcher, kvStore *kv, kvWatchOptions *opts)
 
 natsStatus
 kvStore_Keys(kvKeysList *list, kvStore *kv, kvWatchOptions *opts)
```

Comment:
I would have made this one the static and add the filters/numFilters param. Would have moved the 2 public APIs after it, so no need for forward declaration.

Reply:
My concern is changing the signature of this older function kvStore_Keys would cause a backward incompatible change by breaking the contract.
Let me know if you think otherwise.

Reply:
I did not mean to change the public signature. I would simply have kept the body of this function but made it static, then have `kvStore_Keys()` and the new one invoke the static one. It is simply that the static would have the same params than the one with filters because it can do both. Let me know if that is still not clear.

Reply:
Basically, the changes in `kv.c` could be as simple as that (diffs from main):
```
diff --git a/src/kv.c b/src/kv.c
index 2b079680..09727045 100644
--- a/src/kv.c
+++ b/src/kv.c
@@ -1176,8 +1176,8 @@ kvStore_WatchAll(kvWatcher **new_watcher, kvStore *kv, kvWatchOptions *opts)
     return NATS_UPDATE_ERR_STACK(s);
 }
 
-natsStatus
-kvStore_Keys(kvKeysList *list, kvStore *kv, kvWatchOptions *opts)
+static natsStatus
+_kvStore_Keys(kvKeysList *list, kvStore *kv, kvWatchOptions *opts, const char **filters, int numFilters)
 {
     natsStatus      s;
     kvWatchOptions  o;
@@ -1200,7 +1200,10 @@ kvStore_Keys(kvKeysList *list, kvStore *kv, kvWatchOptions *opts)
     if (o.Timeout > 0)
         timeout = o.Timeout;
 
-    s = kvStore_WatchAll(&w, kv, &o);
+    if (numFilters > 0)
+        s = kvStore_WatchMulti(&w, kv, filters, numFilters, &o);
+    else
+        s = kvStore_WatchAll(&w, kv, &o);
     if (s != NATS_OK)
         return NATS_UPDATE_ERR_STACK(s);
 
@@ -1242,6 +1245,25 @@ kvStore_Keys(kvKeysList *list, kvStore *kv, kvWatchOptions *opts)
     return NATS_UPDATE_ERR_STACK(s);
 }
 
+natsStatus
+kvStore_Keys(kvKeysList *list, kvStore *kv, kvWatchOptions *opts)
+{
+    natsStatus s = _kvStore_Keys(list, kv, opts, NULL, 0);
+    return NATS_UPDATE_ERR_STACK(s);
+}
+
+natsStatus
+kvStore_KeysWithFilters(kvKeysList *list, kvStore *kv, kvWatchOptions *opts, const char **filters, int numFilters)
+{
+    natsStatus s;
+
+    if ((filters == NULL) || (numFilters <= 0))
+        return nats_setDefaultError(NATS_INVALID_ARG);
+
+    s = _kvStore_Keys(list, kv, opts, filters, numFilters);
+    return NATS_UPDATE_ERR_STACK(s);
+}
+
 void
 kvKeysList_Destroy(kvKeysList *list)
 {
```

Reply:
Thanks for the explanation. my assumption was that you wanted to keep the name of the internal method same as kvStore_Keys() (that is why I was concerned about changing the signature as c doesn't support function overloading).